### PR TITLE
HOTT-1379 commodities history behaviour

### PR DESCRIPTION
**Pull request by jebw**
### Jira link

[HOTT-1379](https://transformuk.atlassian.net/browse/HOTT-1379)

### What?

I have added/removed/altered:

- [x] Dropped the automatic redirects for the search since they are too simplistic
- [x] Redirect subheadings from the commodities endpoint

### Why?

I am doing this because:

- We want to match the search and navigation patterns described in the 'scenarios' document attached to the Jira ticket linked above

### CODE REVIEW

File: app/controllers/commodities_controller.rb [hunk 1]
```diff
@@ -1,7 +1,13 @@
 class CommoditiesController < GoodsNomenclaturesController
+  rescue_from Faraday::ResourceNotFound, with: :show_validity_periods
+
   helper_method :uk_declarable, :xi_declarable, :declarable
 
   def show
+    unless declarable.declarable?
+      return redirect_to polymorphic_path(declarable)
```
Comment (willfish) [inline app/controllers/commodities_controller.rb:8]:
```ruby
      def find_commodity
        @commodity = Commodity.actual
                              .declarable
                              .by_code(params[:id])
                              .eager(:goods_nomenclature_indents, :goods_nomenclature_descriptions, :footnotes)
                              .take

        raise Sequel::RecordNotFound if commodity_has_children?
        raise Sequel::RecordNotFound if @commodity.goods_nomenclature_item_id.in? HiddenGoodsNomenclature.codes
      end
```

File: app/controllers/commodities_controller.rb [hunk 2]
```diff
@@ -1,7 +1,13 @@
 class CommoditiesController < GoodsNomenclaturesController
+  rescue_from Faraday::ResourceNotFound, with: :show_validity_periods
+
   helper_method :uk_declarable, :xi_declarable, :declarable
 
   def show
+    unless declarable.declarable?
```
Comment (willfish) [inline app/controllers/commodities_controller.rb:7]: So this is for headings that might not be declarable, right?

Comment (willfish) [reply app/controllers/commodities_controller.rb:7]: AFAIK the commodities controller never returns non-declarable subheadings

File: app/models/subheading.rb [hunk 3]
```diff
@@ -38,6 +38,10 @@ def to_s
     formatted_description || description
   end
 
+  def declarable?
```
Comment (willfish) [inline app/models/subheading.rb]: :tada:

Comment (willfish) [reply app/models/subheading.rb]: I believe this comes back in the api but why not

Comment (willfish) [reply app/models/subheading.rb]: Actually, should we just let the api tell us this? Every goods nomenclature does

Comment (willfish) [reply app/models/subheading.rb]: https://github.com/trade-tariff/trade-tariff-backend/blob/main/app/serializers/api/v2/subheadings/subheading_serializer.rb#L18-L20

File: spec/models/search_spec.rb [hunk 4]
```diff
@@ -90,72 +90,4 @@
       it { is_expected.to be true }
     end
   end
-
-  describe '#search_term_is_commodity_code?' do
```
Comment (willfish) [inline spec/models/search_spec.rb:94]: Wow, actually tested :laughing:

**Comment by willfish** [comment]
Looks awesome. Small comment about duplicating the declarable concept

**Review by willfish — APPROVED** [review]

File: app/models/subheading.rb (hunk 3 above)
Comment (jebw) [reply app/models/subheading.rb]: Ah, ok - will drop this

File: spec/models/search_spec.rb (hunk 4 above)
Comment (jebw) [reply spec/models/search_spec.rb:94]: :laughing: I tend to aim for 100%, sincerely doubt I get it - more then anything it avoids me typoing something then scratching my head why tests further up the stack fail in odd places

Comment (jebw) [reply spec/models/search_spec.rb:94]: tbh if that whole describe block took 5 minutes I'd be surprised, wrote one bit then cut n pasted lots, should probably have used a shared behaviour in hindsight

Comment (willfish) [reply spec/models/search_spec.rb:94]: :tada:

**Pull request merged**